flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/expressions/LookupCallExpression.java

```diff
@@ -43,7 +42,7 @@
 
 	LookupCallExpression(String unresolvedFunction, List<Expression> args) {
 		this.unresolvedName = Preconditions.checkNotNull(unresolvedFunction);
-		this.args = Collections.unmodifiableList(new ArrayList<>(Preconditions.checkNotNull(args)));
+		this.args = Collections.unmodifiableList(Preconditions.checkNotNull(args));
```

Comment:
In theory one can still modify the list from the outside. But I don't have a string opinion here.

Reply:
Honestly I don't know if it makes sense to be so defensive here. 

Usually the arguments are constructed either with `Arrays.asList` or via stream API which produce unmodifiable lists anyway.